networking_ccloud/ml2/agent/common/messages.py

```diff
@@ -341,6 +399,11 @@ def add_binding_host_to_config(self, hg_config, network_id, seg_vni, seg_vlan, t
                 scu.add_vlan(seg_vlan, network_id)
                 scu.add_vxlan_map(seg_vni, seg_vlan)
 
+            # gateways
+            if gateways:
+                scu.add_vlan_iface(vlan=seg_vlan, vrf=gateways['vrf'], primary_ip=gateways['ips'][0],
```

Comment:
We need at least a semi-stable way for the ordering of primary and secondary IPs. In this implmentation the order comes down all the way from the DB. With all the joins inbetween I don't think the order is stable. An unstable order leads to us swapping primary and secondary IP over and over, each time creating a hickup. We probably cannot prevent this completely but relying on the DB feels dangerous.

Reply:
The method that fetches the gateways already sorts them by int value of their network ip, so we always have a stable ordering. Whenever the ordering changes we will reconfigure all ips and the ip with the lowest int value will be the primary ip.